doc/source/data/dataset.rst

```diff
@@ -377,7 +377,149 @@ Tensor datasets can also be created from NumPy ndarrays that are already stored
     arr2 = np.arange(10, 20)
     ds = ray.data.from_numpy([ray.put(arr1), ray.put(arr2)])
 
-Limitations: currently tensor-typed values cannot be nested in tabular records (e.g., as in TFRecord / Petastorm format). This is planned for development.
+Tables with tensor columns
```

Comment:
Shall we split tensor support into a second page?

Reply:
I think that we should start splitting a good bit of this landing page up, but I was hoping to save that for a docs polish PR.

Reply:
Can we specifically break just tensors support out into the next page?